### PR TITLE
Rename rawsockets module to nativesockets

### PR DESCRIPTION
**Pull request by nanoant**
This change was done to avoid confusion with TCP/IP raw sockets. Native sockets
module represents handling native system low level socket API in general and is
not just limited anyhow to TCP/IP raw sockets.

### CODE REVIEW

**Comment by reactormonk** [comment]
You forgot to rename `tests/cpp/trawsockets.nim`. Also, no deprecation of `import rawsockets`?

**Comment by nanoant** [comment]
@reactormonk Oh, okay. As for deprecation shall I leave empty `rawsockets` raising error when imported, or `rawsockets` module including `nativesockets`, emitting warning and having wrappers for renamed procs, eg. `newRawSocket` -> `newNativeSocket`.

**Comment by nanoant** [comment]
@Araq @dom96 This has been rebased onto #3389 PR `patch/separate-deprecated-modules`. So please merge #3389 first.

**Comment by nanoant** [comment]
@dom96 Rebased to latest `devel`, now it should merge fine.

File: lib/pure/nativesockets.nim [hunk 1]
```diff
@@ -231,17 +233,17 @@ proc ntohs*(x: int16): int16 =
   when cpuEndian == bigEndian: result = x
   else: result = (x shr 8'i16) or (x shl 8'i16)
 
-proc htonl*(x: int32): int32 =
+template htonl*(x: int32): expr =
```
Comment (dom96) [inline lib/pure/nativesockets.nim]: Would proc + ``{.inline.}`` be more appropriate here?

Comment (nanoant) [inline lib/pure/nativesockets.nim]: Maybe, but I remember speaking with @Araq and I think using templates is encouraged for doing simple inlining in Nim, and `proc {.inline.}` is for some more complicated cases.

Comment (dom96) [inline lib/pure/nativesockets.nim]: Alright. Merging :)

**Pull request merged**